### PR TITLE
support image prune

### PR DESCRIPTION
**Pull request by junnplus**
Signed-off-by: Ye Sijun <junnplus@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -987,6 +988,19 @@ Flags:
 - :whale: `-q, --quiet`: Only display snapshots IDs
 - :whale: `--format`: Format the output using the given Go template, e.g, `{{json .}}`
 
+### :whale: nerdctl image prune
+Remove unused images.
+
+NOTE: The default behavior of this command is to remove all unused images.
+Inconsistent with docker image prune command, which by default removes only dangling iamge unless the --all flag is specified.
```
Comment (AkihiroSuda) [inline README.md:995]: Better to require `--all` then?

When `--all` is not specified, probably this command should do NOP with a warning message

Comment (junnplus) [reply README.md:995]: SGTM

File: cmd/nerdctl/image_prune.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	defer cancel()
+
+	all, err := cmd.Flags().GetBool("all")
+	if err != nil {
+		return err
+	}
+
+	if !all {
+		logrus.Warn("--all must be specified")
+		// NOP
```
Comment (AkihiroSuda) [inline cmd/nerdctl/image_prune.go:58]: better to warn that this command is (currently) NOP when --all is not specified

Comment (junnplus) [reply cmd/nerdctl/image_prune.go:58]: Sorry, I do not get your point.

Comment (AkihiroSuda) [reply cmd/nerdctl/image_prune.go:58]: like `logrus.Warn("Currently `nerdctl image prune` requires `--all` to be specified. Skipping pruning.")`

File: cmd/nerdctl/image_prune.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	if err != nil {
+		return err
+	}
+	usedImages := make(map[string]struct{})
+	for _, container := range containerList {
+		usedImages[container.Image] = struct{}{}
+	}
+
+	delOpts := []images.DeleteOpt{images.SynchronousDelete()}
+	for _, image := range imageList {
```
Comment (fahedouch) [inline cmd/nerdctl/image_prune.go:98]: I believe that an `imagewalker` (https://github.com/containerd/nerdctl/blob/master/pkg/idutil/imagewalker/imagewalker.go#L46)  is more efficient than this loop

Comment (junnplus) [reply cmd/nerdctl/image_prune.go:98]: Why, can't filter here.

Comment (fahedouch) [reply cmd/nerdctl/image_prune.go:98]: cc @AkihiroSuda  containerd filters are just a loop that compare with inputs ?

Comment (AkihiroSuda) [reply cmd/nerdctl/image_prune.go:98]: Not sure what is to be filtered

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**